### DISCUSSION TITLE
How to compile for Windows x86 (32bits) ?

### DISCUSSION THREAD

**Discussion by Unviray** [sircharlo/meeting-media-manager#585] [Q&A] [answered]
How to compile for Windows x86 (32bits) ?

**Comment by mtdvlpr** [comment]
Do you mean run M3 locally?

**Reply by Unviray** [reply]
No,
My congregation's computer use 32bits version of windows and don't support the version in [Releases](https://github.com/sircharlo/meeting-media-manager/releases)

**Reply by mtdvlpr** [reply]
Ah, okay. Let me see what I can do

**Reply by mtdvlpr** [reply]
Could you try installing [this version](https://drive.google.com/file/d/1ciEy7BGYsD0iZYW0Q874SjBde7TVnYOJ/view?usp=share_link)?

**Reply by mtdvlpr** [reply]
If this works correctly, win32 support will be added from now on.

**Reply by Unviray** [reply]
> Could you try installing [this version](https://drive.google.com/file/d/1ciEy7BGYsD0iZYW0Q874SjBde7TVnYOJ/view?usp=share_link)?

An error occurred while attempting to fetch the following publication: `lff-0`

**Reply by mtdvlpr** [reply]
Hmm, okay. I gave you the latest version (not an official release), so there might be some bug there. But the application is installed successfully?

**Reply by Unviray** [reply]
Yes

**Reply by mtdvlpr** [reply]
Alright. Then this will be included from now on. And I will look into the `lff-0` issue.

**Reply by sircharlo** [reply]
If I recall correctly, one of the dependencies doesn't have a 32-bit version...

**Reply by mtdvlpr** [reply]
Hmm, we did change all the dependencies since v22.8.1.

**Comment by mtdvlpr** [comment]
@Unviray, could you try [this version](https://drive.google.com/file/d/16q8lwozrgawFv5JQYpNHpWuKS4l5s8Gb/view?usp=share_link) and if an error occurs, open the console tab and paste the error(s) here?

**Reply by Unviray** [reply]
How to run it from console ?
I'm not familiar with Windows

**Reply by mtdvlpr** [reply]
You don't have to run it from console. But when it's open, press CTRL + SHIFT + i

**Reply by Unviray** [reply]
`instrument.js:108`
```
TypeError: The "path" argument must be of type string or an instance of Buffer or URL. Received undefined
    at __node_internal_captureLargerStackTrace (node:internal/errors:465:5)
    at new NodeError (node:internal/errors:372:5)
    at __node_internal_ (node:internal/fs/utils:671:11)
    at __node_internal_ (node:internal/fs/utils:683:3)
    at Object.openSync (node:fs:577:10)
    at Object.func [as openSync] (node:electron/js2c/asar_bundle:5:1810)
    at readFileSync (node:fs:453:35)
    at e.readFileSync (node:electron/js2c/asar_bundle:5:9156)
    at _ (media.ts:675:19)
    at async media.ts:79:13
    at async Promise.allSettled (index 3)
    at async F (media.ts:175:5)
    at async f.<anonymous> (media.ts:815:10)
    at async f.getMwMedia (index.vue?0145:394:9)
    at async Promise.allSettled (index 0)
    at async f.startMediaSync (index.vue?0145:512:29)
```

**Reply by mtdvlpr** [reply]
If you change the `Default levels` value at the menu bar of the console to also check the`Verbose` you should see more.

**Reply by mtdvlpr** [reply]
![image](https://user-images.githubusercontent.com/46671786/199966700-a892a8b5-6be6-4cf1-8c06-6f5b13eb4af0.png)

**Reply by Unviray** [reply]
There are too many logs
[1667563440441.log](https://github.com/sircharlo/meeting-media-manager/files/9938083/1667563440441.log)

**Reply by mtdvlpr** [reply]
Thanks! I fixed it and added more logs around it. [This version](https://drive.google.com/file/d/10E_Ak_XEK7NxoSMxu9OwRd6hxTpyzGYE/view?usp=share_link) should provide more information on the problem.

**Reply by mtdvlpr** [reply]
`Verbose` should not be necessary for this one.

**Reply by Unviray** [reply]
```
pub lff
issue 0
localpath
```
```
No db file found in pubPath C:/Users/LENOVO/AppData/Roaming/meeting-media-manager/Publications/MG/lff/0/0
```

**Reply by mtdvlpr** [reply]
Does that local path exist? `C:/Users/LENOVO/AppData/Roaming/meeting-media-manager/Publications/MG/lff/0/0`. And if so, what is in it?

**Comment by Unviray** [comment]
Yes, it exists

There is an lff book in it (lff_MG.jwpub)

Le ven. 4 nov. 2022, 18:14, Manoah Tervoort ***@***.***> a
écrit :

**Reply by mtdvlpr** [reply]
That's the only file in it?

**Reply by sircharlo** [reply]
Sounds to me like the unzipper isn't doing its job...

**Comment by Unviray** [comment]
Yes

Le ven. 4 nov. 2022, 18:22, Manoah Tervoort ***@***.***> a
écrit :

**Reply by mtdvlpr** [reply]
Alright. Then I found the culprit. We're using a dependency to extract the media from a jwpub file. That folder should contain a bunch of images and a .db file. I will try to find an alternative package that supports win32. Thank you for being so patient while figuring this out!

**Reply by sircharlo** [reply]
I'm trying to set up a virtual 32-bit version of Windows 10 as we speak to run some tests, if ever it finishes installing.

**Comment by Unviray** [comment]
Thank you for you support.

I extract it manually and it works

**Comment by mtdvlpr** [comment]
[This version](https://drive.google.com/file/d/1Kx1VxVSpXI6eQUMoPXXiWMr-zaMHflPF/view?usp=share_link) is a potential fix. I'm now using another, more mature package.

**Reply by mtdvlpr** [reply]
Make sure to remove your cache via the settings before fetching.

**Reply by sircharlo** [reply]
Confirmed as working on my fresh Windows 32-bit installation! Good job man

![image](https://user-images.githubusercontent.com/1275666/200030327-9662e290-b1f7-4775-af5b-bddbd8124c7e.png)

**Reply by mtdvlpr** [reply]
Alright, I also implemented the new JSZip in other places.

**Comment by mtdvlpr** [comment, marked as answer]
Added in #608. Will be in the next release :)

**Comment by mtdvlpr** [comment]
Hi @Unviray, I wanted to check in to see if the 32-bit version works as expected for you?

**Reply by Unviray** [reply]
Hi,
Yes, it works
Thank you

**Reply by mtdvlpr** [reply]
Great to hear! Just a small reminder, the translations for Malagasy are at 22%. No rush, take your time. Just a reminder that even though we have already added Malagasy as a language, it's not fully translated yet. If you encounter any issues while translating, let us know!
